content/Link Tracking with StimulusReflex - Part III.md

```diff
@@ -0,0 +1,156 @@
+get redirected to the original URL and we'll track one morer view for that link.
+
+### Displaying link views
+
+At this point we could just make a static page and spit out our data, but what's the fun in that?
+As a way to show how to pass data from the server to the view, as opposed to what we've been doing until now,
+we'll make page views update in real time using CableReady. 
+
+To do this, let's first generate a new channel by running:
+
+```bash
+bundle exec rails generate channel LinkViews
+```
+
+Then we'll change `app/channels/link_views_channel.rb` to just stream from `"link_views"`
+
+```ruby
+class LinkViewsChannel < ApplicationCable::Channel
+  def subscribed
+    stream_from "link_views"
+  end
+end
+```
+
+We must now tell our client channel, the Javascript one, to perform CableReady operations
+if there are any. You can do that by editing the code inside
+`app/javascript/channels/link_views_channel.js` to match the following snippet:
+
+```javascript
+consumer.subscriptions.create('LinkViewsChannel', {
+  received(data) {
+    if (data.cableReady) {
+      CableReady.perform(data.operations)
+    }
+  }
+});
+```
+
+And don't forget to actually import CableReady at the top of the file
+
+```javascript
+import CableReady from 'cable_ready'
+```
+
+Let's change our Rails views now to actually show the link's views count.
+Go to `app/views/shortened_links/index.html.erb` and add a new column on the table.
+
+
+```erb
+  <div class="col"><b>Original url</b></div>
+  <div class="col"><b>Views</b></div>
+  <div class="col"></div>
+```
+
+And populate it with the right data:
+
+```erb
+  <div class="col"><%= shortened_link.original_url %></div>
+  <div class="col"><%= tag.span shortened_link.views_count, id: "link-#{shortened_link.id}-views-count" %></div>
+  <div class="col">
+```
+
+Notice how we wrapped the views count inside a span tag with an id.
+We are doing this because we need to tell CableReady to perform operations
+later on and for that we require a selector.
+
+So just copy the selector and change `app/controllers/redirections_controller.rb` to look like this
+
+```ruby
+class RedirectionsController < ApplicationController
+  include CableReady::Broadcaster
+
+  def index
+    link = ShortenedLink.find_by!(shortened_path: params[:path])
+    link.increment!(:views_count)
+
+    cable_ready["link_views"].text_content(
+      selector: "#link-#{link.id}-views-count",
+      text: link.views_count
+    )
+    cable_ready.broadcast
+
+    redirect_to link.original_url
+  end
+end
+```
+
+What's going on here? Well it's pretty simple, we are including `CableReady::Broadcaster`
+to be able to broadcast operations. Then we add an operation on the 
+`"link_views"` channel, you know the one we set to stream from before. The operation
+is a text replacement in the node with the id `link-#{link.id}-views-count`,
+that's what the `#` symbolizes, and the text is the new views count.
+
+### Conclusion
```

Comment:
maybe a better conclusion since it's the last part? like a recap of everything that was achieved?